src/extensions/default/Inline3DParametersEditor/ParameterEditor.js

```diff
@@ -0,0 +1,149 @@
+define(function (require, exports, module) {
+    "use strict";
+
+    var Strings            = brackets.getModule("strings"),
+        Mustache           = brackets.getModule("thirdparty/mustache/mustache");
+
+    /** Mustache template that forms the bare DOM structure of the UI */
+    var Template = require("text!ParameterEditorTemplate.html");
+
+     //used to control the smoothness of scroll.
+     var MODIFIER = 10;
+
+    function ParameterEditor($parent, callback) {
+        // Create the DOM structure, filling in localized strings via Mustache
+        this.$element = $(Mustache.render(Template, Strings));
+        $parent.append(this.$element);
+
+        this._callback = callback;
+
+        this._handleXValueSliderDrag = this._handleXValueSliderDrag.bind(this);
+        this._handleYValueSliderDrag = this._handleYValueSliderDrag.bind(this);
+        this._handleZValueSliderDrag = this._handleZValueSliderDrag.bind(this);
+        this.$xValueSlider = this.$element.find("#_x");
+        this.$yValueSlider = this.$element.find("#_y");
+        this.$zValueSlider = this.$element.find("#_z");
+        this._x = parseFloat(this.$xValueSlider.val());
+        this._y = parseFloat(this.$yValueSlider.val());
+        this._z = parseFloat(this.$zValueSlider.val());
+
+        this.$selectionBase = this.$element.find(".selector-base");
+
+        // Attach event listeners to main UI elements
+        this._addListeners();
+    }
+
+
+    /** Returns the root DOM node of the UI */
+    ParameterEditor.prototype.getRootElement = function () {
+        return this.$element;
+    };
+
+    ParameterEditor.prototype._addListeners = function () {
+        var self = this;
+        this.$xValueSlider.mousedown(function(e) {
+            self._x = parseFloat(e.currentTarget.value);
+            self._position = e.clientX;
+        });
+
+        this.$yValueSlider.mousedown(function(e) {
+            self._y = parseFloat(e.currentTarget.value);
+            self._position = e.clientX;
+        });
+
+        this.$zValueSlider.mousedown(function(e) {
+            self._z = parseFloat(e.currentTarget.value);
+            self._position = e.clientX;
+        });
+
+        
+        this.$xValueSlider.change(function(e) {
+            self._commitParameters(self.$xValueSlider.val() + " 3 0");
```

Comment:
Was testing when I was working with the x direction only. Forgot to edit it.
Will fix it